worlds/subnautica/Rules.py

```diff
@@ -1,122 +1,128 @@
+def has_vehicle_upgrade_console(state: "CollectionState", player: int):
     return state.has("Vehicle Upgrade Console", player) and \
            has_moonpool(state, player)
 
 
-def has_seamoth(state, player: int):
+def has_seamoth(state: "CollectionState", player: int):
     return state.has("Seamoth Fragment", player, 3) and \
            has_mobile_vehicle_bay(state, player)
 
 
-def has_seamoth_depth_module_mk1(state, player: int):
+def has_seamoth_depth_module_mk1(state: "CollectionState", player: int):
     return has_vehicle_upgrade_console(state, player)
 
 
-def has_seamoth_depth_module_mk2(state, player: int):
+def has_seamoth_depth_module_mk2(state: "CollectionState", player: int):
     return has_seamoth_depth_module_mk1(state, player) and \
            has_modification_station(state, player)
 
 
-def has_seamoth_depth_module_mk3(state, player: int):
+def has_seamoth_depth_module_mk3(state: "CollectionState", player: int):
     return has_seamoth_depth_module_mk2(state, player) and \
            has_modification_station(state, player)
 
 
-def has_cyclops_bridge(state, player: int):
+def has_cyclops_bridge(state: "CollectionState", player: int):
     return state.has("Cyclops Bridge Fragment", player, 3)
 
 
-def has_cyclops_engine(state, player: int):
+def has_cyclops_engine(state: "CollectionState", player: int):
     return state.has("Cyclops Engine Fragment", player, 3)
 
 
-def has_cyclops_hull(state, player: int):
+def has_cyclops_hull(state: "CollectionState", player: int):
     return state.has("Cyclops Hull Fragment", player, 3)
 
 
-def has_cyclops(state, player: int):
+def has_cyclops(state: "CollectionState", player: int):
     return has_cyclops_bridge(state, player) and \
            has_cyclops_engine(state, player) and \
            has_cyclops_hull(state, player) and \
            has_mobile_vehicle_bay(state, player)
 
 
-def has_cyclops_depth_module_mk1(state, player: int):
+def has_cyclops_depth_module_mk1(state: "CollectionState", player: int):
     return state.has("Cyclops Depth Module MK1", player) and \
            has_modification_station(state, player)
 
 
-def has_cyclops_depth_module_mk2(state, player: int):
+def has_cyclops_depth_module_mk2(state: "CollectionState", player: int):
     return has_cyclops_depth_module_mk1(state, player) and \
            has_modification_station(state, player)
 
 
-def has_cyclops_depth_module_mk3(state, player: int):
+def has_cyclops_depth_module_mk3(state: "CollectionState", player: int):
     return has_cyclops_depth_module_mk2(state, player) and \
            has_modification_station(state, player)
 
 
-def has_prawn(state, player: int):
+def has_prawn(state: "CollectionState", player: int):
     return state.has("Prawn Suit Fragment", player, 4) and \
            has_mobile_vehicle_bay(state, player)
 
 
-def has_praw_propulsion_arm(state, player: int):
+def has_praw_propulsion_arm(state: "CollectionState", player: int):
```

Comment:
```suggestion
def has_prawn_propulsion_arm(state: "CollectionState", player: int):
```

Needs refactor to fix misspelling.